src/Microsoft.Dnx.ApplicationHost/Program.cs

```diff
@@ -242,23 +253,21 @@ private Task<int> ExecuteMain(DefaultHost host, string applicationName, string[]
             return EntryPointExecutor.Execute(assembly, args, host.ServiceProvider);
         }
 
-        private static void ThrowEntryPointNotfoundException(
+        private static void NotifyEntryPointNotfound(
             DefaultHost host,
             string applicationName,
             Exception innerException)
         {
             if (host.Project.Commands.Any())
             {
-                // Throw a nicer exception message if the command
-                // can't be found
-                throw new InvalidOperationException(
-                    string.Format("Unable to load application or execute command '{0}'. Available commands: {1}.",
+                Console.Error.WriteLine(string.Format("Unable to load application or execute command '{0}'. Available commands: {1}.",
```

Comment:
So we want to swallow all user exceptions? There are benefits to allowing actual exceptions from user code to bubble up to Watson, though I suppose those are most relevant when the app has been packaged separately as it's own executable and not `dnx.exe`. The initial thought here was to just capture exceptions caused by our code and allow exceptions caused by user code to bubble.

On a side note, maybe the error handling you referenced above is why I can never get unhandled exceptions to break in the debugger... I don't know if it works for users, but I can't ever get them to work.